qiskit/circuit/library/data_preparation/initializer.py

```diff
@@ -0,0 +1,96 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Initialize qubit registers to desired arbitrary state.
+"""
+
+from __future__ import annotations
+from collections.abc import Sequence
+import typing
+
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit.quantumregister import QuantumRegister
+from qiskit.circuit.instruction import Instruction
+from .state_preparation import StatePreparation
+
+if typing.TYPE_CHECKING:
+    from qiskit.quantum_info.states.statevector import Statevector
+
+_EPS = 1e-10  # global variable used to chop very small numbers to zero
+
+
+class Initialize(Instruction):
+    """Complex amplitude initialization.
+
+    Class that initializes some flexible collection of qubit registers, implemented by calling
+    the :class:`qiskit.extensions.StatePreparation` Class.
```

Comment:
Should `qiskit.extensions` here also be updated?

Reply:
Good catch, yes it should!